src/au3wrap/internal/au3audiodevicesprovider.cpp

```diff
@@ -21,11 +23,141 @@
 using namespace muse;
 using namespace au::au3;
 
+static const muse::Settings::Key AUDIO_HOST("au3wrap", "AudioIO/Host");
+static const muse::Settings::Key PLAYBACK_DEVICE("au3wrap", "AudioIO/PlaybackDevice");
+static const muse::Settings::Key RECORDING_DEVICE("au3wrap", "AudioIO/RecordingDevice");
+static const muse::Settings::Key INPUT_CHANNELS("au3wrap", "AudioIO/RecordChannels");
+
+static const muse::Settings::Key LATENCY_DURATION("au3wrap", "AudioIO/LatencyDuration");
+static const muse::Settings::Key LATENCY_COMPENSATION("au3wrap", "AudioIO/LatencyCorrection");
+
+static const muse::Settings::Key DEFAULT_SAMPLE_RATE("au3wrap", "SamplingRate/DefaultProjectSampleRate");
+static const muse::Settings::Key SAMPLE_FORMAT("au3wrap", "SamplingRate/DefaultProjectSampleFormatChoice");
+
 void Au3AudioDevicesProvider::init()
 {
     initHosts();
     initHostDevices();
+
+    std::string defaultHost = !audioApiList().empty() ? audioApiList()[0] : "";
```

Comment:
better to use front

Reply:
Also, defining the default device should be more complicated than just selecting the first device from the list. 
I see that DeviceManager has GetDefaultOutputDevice, but I don't know what hostIndex means. 
In MuseScore, for example, we have the concept of "default" deviceId, and depending on this value, we establish a connection differently.